Userland/Applications/PixelPaint/Layer.cpp

```diff
@@ -55,8 +55,9 @@ RefPtr<Layer> Layer::create_with_bitmap(Image& image, const Gfx::Bitmap& bitmap,
 RefPtr<Layer> Layer::create_snapshot(Image& image, const Layer& layer)
 {
     auto snapshot = create_with_bitmap(image, *layer.bitmap().clone(), layer.name());
-    snapshot->set_opacity_percent(layer.opacity_percent());
-    snapshot->set_visible(layer.is_visible());
+    snapshot->m_opacity_percent = layer.opacity_percent();
+    snapshot->m_visible = layer.is_visible();
+
     snapshot->set_selected(layer.is_selected());
     snapshot->set_location(layer.location());
```

Comment:
It's not obvious while reading this code why some things are assigned directly and others go through setters. At the very least let's add a comment explaining that these must be set directly to avoid notifying the `Image`.

Reply:
Done